corehq/blobs/tests/test_export.py

```diff
@@ -1,123 +1,90 @@
-        for doc_class, domain, data in files:
-            obj = doc_class.get_by_data(data)
-            obj.attach_data(data)
-            obj.add_domain(domain)
-            self.addCleanup(obj.delete)
-            self.assertEqual(data, obj.get_display_file(False))
-            blob_keys.append(obj.blobs[obj.attachment_id].key)
+    def test_orphaned_blob_meta_is_not_exported(self):
+        expected_meta = self.db.put(
+            BytesIO(self.blob_data), meta=new_meta(domain=self.domain, type_code=CODES.form_xml)
+        )
+        self.addCleanup(self.db.delete, expected_meta.key)
+        orphaned_meta = new_meta(domain=self.domain, type_code=CODES.form_xml, content_length=42)
+        orphaned_meta.save()
 
-        expected = set(blob_keys[:-1])
         with NamedTemporaryFile() as out:
-            exporter = EXPORTERS['multimedia'](self.domain_name)
-            exporter.migrate(out.name, force=True)
+            self.exporter.migrate(out.name, force=True)
             with tarfile.open(out.name, 'r:gz') as tgzfile:
-                self.assertEqual(expected, set(tgzfile.getnames()))
+                self.assertCountEqual({expected_meta.key}, tgzfile.getnames())
 
-    def test_export_and_import_compressed_blobs(self):
+    def test_exported_blobs_can_be_imported_successfully(self):
+        form = self.db.put(BytesIO(self.blob_data), meta=new_meta(domain=self.domain, type_code=CODES.form_xml))
         with NamedTemporaryFile() as out:
-            exporter = EXPORTERS['all_blobs'](self.domain_name)
-            exporter.migrate(out.name, force=True)
-            self.import_and_verify(out.name)
-
-    def import_and_verify(self, filename):
-        from ..management.commands.run_blob_import import Command as ImportCommand
-        expected_metas = {m for m in self.blob_metas
-            if m.key not in self.not_found and m.domain == self.domain_name}
-        self.assertTrue(any(m.is_compressed for m in expected_metas))
-        with TemporaryFilesystemBlobDB() as dest_db:
-            assert get_blob_db() is dest_db, (get_blob_db(), dest_db)
-            ImportCommand.handle(None, filename)
-            for meta in expected_metas:
-                with dest_db.get(meta=meta) as fh:
-                    self.assertEqual(fh.read(), self.data, meta.type_code)
+            self.exporter.migrate(out.name, force=True)
+            with TemporaryFilesystemBlobDB() as dest_db:
+                assert get_blob_db() is dest_db, (get_blob_db(), dest_db)
+                ImportCommand.handle(None, out.name)
+                with dest_db.get(meta=form) as fh:
+                    self.assertEqual(fh.read(), self.blob_data)
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain = 'exporter-test'
+        cls.blob_data = b'binary data not valid utf-8 \xe4\x94'
+        cls.exporter = BlobExporter(cls.domain)
+
+    def setUp(self):
+        super().setUp()
+        self.db = blob_db()
```

Comment:
Moving these to the end of the class made it difficult to see how they changed.

Reply:
Addressed in b163a09933a06bf665cb817ed2b67e5300e02699